dbt/clients/git.py

```diff
@@ -59,3 +60,32 @@ def get_current_sha(cwd):
 
 def remove_remote(cwd):
     return run_cmd(cwd, ['git', 'remote', 'rm', 'origin'])
+
+
+def clone_and_checkout(repo, cwd, dirname=None, remove_git_dir=False,
+                       branch=None):
+    _, err = clone(repo, cwd, dirname=dirname, remove_git_dir=remove_git_dir)
+    exists = re.match("fatal: destination path '(.+)' already exists",
+                      err.decode('utf-8'))
+    directory = None
+    start_sha = None
+    if exists:
+        directory = exists.group(1)
+        logger.info('Updating existing dependency %s.', directory)
```

Comment:
Yeah I noticed that too, particularly because since I wrote some of the deps code to do the checkout at a few different places (just to make the code cleaner) it will always print the repo is already up to date. I can change all the info logs in this module to debug unless you think there are any in particular that should be info.

Reply:
Yeah, i think that's a good idea. We'll still want to show some info to the user with info-level logging, but i don't think any of that ui-related code belongs in the git client